ADR 006: Code-Outward Documentation (#79)
pyx-industries/julee: docs/ADRs/006-code-outward-documentation.md

# ADR 006: Code-Outward Documentation

## Status

Draft

## Date

2025-12-28

## Context

Julee documentation exists in parallel forms: hand-written RST and autodoc-generated
API docs. This creates drift, duplication, and maintenance burden.

The doctrine system (ADR 002) establishes that tests ARE the specification. The same

> principle applies to documentation: **docstrings ARE the documentation**.

Reviewer: Is it all based on python docstrings only? If not, maybe:
```suggestion
principle applies to documentation: **the code documentation IS the documentation**.
```
though that doesn't quite capture it. Maybe `docstrings` is fine - by which we just mean generally high-level (module/function) code documentation, I guess?
